src/include/daos_srv/container.h

```diff
@@ -46,12 +46,14 @@ int ds_cont_tgt_open(uuid_t pool_uuid, uuid_t cont_hdl_uuid,
  * Per-thread container (memory) object
  *
  * Stores per-thread, per-container information, such as the vos container
- * handle.
+ * handle. N.B. sc_uuid and sc_pool_uuid must be contiguous in memory,
+ * used as a 256 bit key in tls dt_cont_cache.
```

Comment:
Should we assert that, say, in cont_child_cmp_keys?

    D_CASSERT(offsetof(struct ds_cont_child, sc_uuid) + sizeof(uuid_t) ==
              offsetof(struct ds_cont_child, sc_pool_uuid));

Reply:
That is a good idea. I am going to add it to the cont_child_alloc_ref() function that seems a natural place for code readability to put that compile-time assert for the format of the structure.